storage/storage.py

```diff
@@ -287,18 +350,41 @@ def delete(self):
         filename = self._cd_to_file(ftp_client)
         ftp_client.delete(filename)
 
+    def get_download_url(self, seconds=60, key=None):
+        """
+        Return a temporary URL allowing access to the storage object.
+
+        If a download_url_base is specified in the storage URI, then a call to get_download_url() will
+        return the download_url_base joined with the object name.
+
+        For example, if "http://www.someserver.com:1234/path/to/" were passed (urlencoded) as the
+        download_url_base query parameter of the storage URI:
+
+          ftp://username:password@hostname/some/path/to/a/file.txt?download_url_base=http%3A%2F%2Fwww
+          .someserver.com%3A1234%2Fpath%2Fto%2
+
+        then a call to get_download_url() would yield:
+
+          http://www.someserver.com:1234/path/to/file.txt
+
+
+        :param seconds:  ignored for ftp storage
+        :param key:      ignored for ftp storage
+        :return:
+        """
+        temp_url = None
```

Comment:
All the same comments as in the file protocol.

Also, this appears to be duplicated code. Can you move it to a helper function?